src/resources/IPAsset.py

```diff
@@ -45,24 +43,108 @@ def register(self, token_contract, token_id):
                 }
 
             # Build the transaction
-            transaction = self.ip_asset_registry_client.contract.functions.register(self.chain_id, token_contract, token_id).build_transaction({
-                'from': self.account.address,
-                'nonce': self.web3.eth.get_transaction_count(self.account.address),
-                'gas': 2000000,
-                'gasPrice': self.web3.to_wei('100', 'gwei')  # Adjusted gas price for faster processing
-            })
+            transaction = self.ip_asset_registry_client.build_register_transaction(
+                self.chain_id, token_contract, token_id, {
+                    'from': self.account.address,
+                    'nonce': self.web3.eth.get_transaction_count(self.account.address),
+                    'gas': 2000000,
+                    'gasPrice': self.web3.to_wei('300', 'gwei')  # Adjusted gas price for faster processing
```

Comment:
can you remove or generalize this? 300 gwei for mainnet is too high

Reply:
Going to add the optn to specify txn options late.r Right now, just setting a default value of 300. I agree that it's too high but when I was trying to register, using 100 wasn't working which is y I increased to 300. Can change all to 100 or even lower but I am unsure if the txns will go thru